### PR TITLE
As/top line

### PR DESCRIPTION
**Pull request by anbitasiregar**

### CODE REVIEW

File: ThreadGroup/Views/TGSelectDeviceStepView.m [hunk 1]
```diff
@@ -65,6 +65,10 @@ - (void)commonInit {
 
 #pragma mark - Public
 
+- (void)setTopBarHidden:(BOOL)hidden {
+    self.topBar.hidden = hidden;
+}
```
Comment (luquantan) [inline ThreadGroup/Views/TGSelectDeviceStepView.m]: I think you could just call `self.topBar.hidden = YES` or `self.topBar.hidden = NO`

Comment (anbitasiregar) [inline ThreadGroup/Views/TGSelectDeviceStepView.m]: Okay, I'll change it! I was following the format of TGDeviceStepView, which does this.

Comment (luquantan) [inline ThreadGroup/Views/TGSelectDeviceStepView.m]: Ah...but its done in TGDeviceStepView because we make the method public instead of the property.

**Comment by luquantan** [comment]
:ship:

File: ThreadGroup/Views/TGSelectDeviceStepView.m (hunk 1 above)
Comment (anbitasiregar) [inline ThreadGroup/Views/TGSelectDeviceStepView.m]: Got it. Should I still change it?

**Comment by ButkiewiczP** [comment]
I think this is actually more work than is necessary.

In the method `- (void)setContentMode:(TGSelectDeviceStepViewContentMode)contentMode` in `TGSelectDeviceStepView` there's a switch that changes a number of properties based on the contentMode that's set.

Right now, we have the `-setTopBarHidden` method inside every single one of these switch cases. The only time a different value is passed is in the `TGSelectDeviceStepViewContentModeComplete` case. Instead of calling this method inside the switch, it'd be better to have a single line outside of the switch that will act in all cases.

For example, we're already doing this exact thing in this function for the `topSeperatorBar` view. At the bottom of that function you'll see 

`self.topSeperatorBar.hidden = (contentMode != TGSelectDeviceStepViewContentModeComplete);` 

Which is exactly what we want, except with the topBar and topSeperatorBar. All you should need to do is copy and modify that line for the `topBar` view. The extra `-setTopBarHidden` function shouldn't be necessary.

**Pull request merged**